### PR TITLE
change controller initialize

### PR DESCRIPTION
**Pull request by dhkim0225**
1. op_tanh_reduce 추가
2. tanh_constant 값 변경
3. bias False 옵션들 추가
4. Xavier initializer 로 layer init 변경
5. action 바로 controller에서 detach()
6. additional_bias __init__으로 옮기고 옵션을 통해 추가할 수 있도록 함 (원 코드 따라하기) (기존 --> 그냥 추가해줬었음)

### CODE REVIEW

File: enas/controller.py [hunk 1]
```diff
@@ -8,62 +8,61 @@
...
-		# print(self.fc_ops.bias.size())
-		init_bias = [10, 10, 0, 0, 0]
-		self.fc_ops.bias.data = torch.Tensor(np.array(init_bias))
-		
+		torch.nn.init.xavier_uniform(self.embed_first.weight)
+		torch.nn.init.xavier_uniform(self.embed_ops.weight)
+		torch.nn.init.xavier_uniform(self.lstm.weight_hh)
+		torch.nn.init.xavier_uniform(self.lstm.weight_ih)
+
+		self.fc_ops.bias.data = torch.Tensor([10, 10, 0, 0, 0])
```
[the remark this answers is not in the document]
Comment (dhkim0225) [reply enas/controller.py]: 'fc_ops' is Parameter, so if model is on gpu then 'fc_ops' will also be on gpu automatically.

File: enas/controller.py [hunk 2]
```diff
@@ -8,62 +8,61 @@
...
-			param.data.uniform_(-init_range, init_range)
-
-		self.fc_index_prev.bias.data.fill_(0)
-		self.fc_index_curr.bias.data.fill_(0)
-		# print(self.fc_ops.weight.size())
-		# print(self.fc_ops.bias.size())
-		init_bias = [10, 10, 0, 0, 0]
-		self.fc_ops.bias.data = torch.Tensor(np.array(init_bias))
-		
+		torch.nn.init.xavier_uniform(self.embed_first.weight)
```
[the remark this answers is not in the document]
Comment (dhkim0225) [reply enas/controller.py]: default initialization of tf.get_variable() function is Xavier Uniform. 
https://stackoverflow.com/questions/37350131/what-is-the-default-variable-initializer-in-tensorflow

File: enas/controller.py [hunk 3]
```diff
@@ -76,13 +75,12 @@ def sample_cell(self, arc_seq, entropy_list, log_prob_list):
...
+				logits = self.tanh_constant * F.tanh(logits)
 				probs = F.softmax(logits, dim=-1)
 				log_prob = F.log_softmax(logits, dim=-1)
-				action = torch.multinomial(probs, 1)[0]
+				action = torch.multinomial(probs, 1)[0].detach()
 				arc_seq.append(action)
-				
+
 				selected_log_prob = log_prob[:, action.long()]
 				entropy = -(log_prob * probs).sum(1, keepdim=False)
```
[the remark this answers is not in the document]
Comment (dhkim0225) [reply enas/controller.py]: I think it may save tiny GPU memory.

File: enas/controller.py [hunk 4]
```diff
@@ -95,16 +93,15 @@ def sample_cell(self, arc_seq, entropy_list, log_prob_list):
 				hidden = (self.hx, self.cx)
 				self.hx, self.cx = self.lstm(inputs, hidden)
 				logits = self.fc_ops(self.hx)
-				logits /= self.temperature
-				logits = self.tanh_constant*F.tanh(logits)
-				init_bias = [0.25, 0.25, -0.25, -0.25, -0.25]
-				logits += torch.Tensor(np.array(init_bias)).to(device)
+				logits = (self.tanh_constant / self.op_tanh_reduce) * F.tanh(logits)
```
[the remark this answers is not in the document]
Comment (dhkim0225) [reply enas/controller.py]: image
Please tell me more about it! Above image is the original code(tf).

File: enas/controller.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dhkim0225) [reply enas/controller.py]: 아 아닌가요...?

File: enas/controller.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (dnddnjs) [reply enas/controller.py]: I bring the code from carpedm20's code. Let's try author's code

File: enas/controller.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (dnddnjs) [reply enas/controller.py]: Then you have to remain detach() in the main function

[the remark this answers is not in the document]
Comment (dhkim0225) [reply enas/controller.py]: I GOT IT!
